tests/test_utils.py

```diff
@@ -24,10 +24,14 @@ def get_bad_input_path(file_name):
 
 
 def generate_ttx_dump(font_path, tables=None):
-    with TTFont(font_path) as font:
-        temp_path = get_temp_file_path()
-        font.saveXML(temp_path, tables=tables)
-        return temp_path
+    try:
+        font = TTFont(font_path)
+    except TTLibError:
+        font = TTCollection(font_path)
+
+    temp_path = get_temp_file_path()
+    font.saveXML(temp_path, tables=tables)
```

Comment:
Please add `font.close()` after `font.saveXML()`.

Reply:
I have investigated this issue before I decided to remove "with" statement. "TTCollection" class does not provide the "close" method. Therefore, adding `font.close()` is impracticable. However, due to garbage collector of Python, the font object will be automatically closed and deleted, but heavily relying on garbage collector is a bad practice. Of course, there are at least two workarounds:
1. Re-implement the close method:
```
if font.reader is not None:
   font.reader.close()
```
2. Open font file by built-in open:
```
with open(font_path, 'rb') as f:    
    try:
        font = TTFont(f)
    except TTLibError:
        font = TTCollection(f)

    temp_path = get_temp_file_path()
    font.saveXML(temp_path, tables=tables)
    return temp_path
```
Nevertheless, both methods, I think, will add  maintenance burden if the "TTCollection" class implement close method in some day.
So, how do you think?

Reply:
I think the best would be to implement a `close()` method in the `TTCollection` class.
Would you mind submitting such change to https://github.com/fonttools/fonttools ?

Reply:
Not at all.

Reply:
Moreover, if I submit such change and it is merged in "fonttools", this whole project will require the latest "fonttools". Hence, is it worthy to do so?

Reply:
Yes. The afdko always updates to the latest version of FontTools.
A lot of other projects rely on FontTools so every opportunity to make the changes there benefits everyone.